openhands/controller/agent_controller.py

```diff
@@ -312,11 +312,13 @@ async def _handle_observation(self, observation: Observation) -> None:
             elif isinstance(observation, ReplayPhaseUpdateObservation):
                 new_phase = observation.new_phase
                 if self.state.replay_phase == new_phase:
-                    raise ValueError(
-                        f'Unexpected ReplayPhaseUpdateAction. Already in phase: {new_phase}'
+                    self.log(
```

Comment:
NOTE: Since we ask it to verify its own hypothesis, it sometimes tries to re-submit it, which takes us here. No reason to throw in that case. (Although it is weird because it technically has no longer access to the tool. Might want to debug why it can even this sometime.)